opentelemetry/proto/profiles/v1/profiles.proto

```diff
@@ -0,0 +1,133 @@
+  string schema_url = 3;
+}
+
+// A collection of Profiles produced by an InstrumentationScope.
+message ScopeProfiles {
+  // The instrumentation scope information for the profiles in this message.
+  // Semantically when InstrumentationScope isn't set, it is equivalent with
+  // an empty instrumentation scope name (unknown).
+  opentelemetry.proto.common.v1.InstrumentationScope scope = 1;
+
+  // A list of Profiles that originate from an instrumentation scope.
+  repeated Profile profiles = 2;
+
+  // This schema_url applies to all profiles and profile events in the "profiles" field.
+  string schema_url = 3;
+}
+
+// A Profile represents a single profile.
+message Profile {
+  // A unique identifier for a profile. The ID is a 16-byte array. An ID with
+  // all zeroes is considered invalid.
+  //
+  // This field is required.
+  bytes profile_id = 1;
+
+  // start_time_unix_nano is the start time of the profile.
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  //
+  // This field is semantically required and it is expected that end_time >= start_time.
+  fixed64 start_time_unix_nano = 2;
+
+  // end_time_unix_nano is the end time of the profile.
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  //
+  // This field is semantically required and it is expected that end_time >= start_time.
+  fixed64 end_time_unix_nano = 3;
+
+  // attributes is a collection of key/value pairs. Note, global attributes
+  // like server name can be set using the resource API. Examples of attributes:
+  //
+  //     "/http/user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36"
+  //     "/http/server_latency": 300
+  //     "abc.com/myattribute": true
+  //     "abc.com/score": 10.239
+  //
+  // The OpenTelemetry API specification further restricts the allowed value types:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute
+  // Attribute keys MUST be unique (it is not allowed to have more than one
+  // attribute with the same key).
+  repeated opentelemetry.proto.common.v1.KeyValue attributes = 4;
+
+  // dropped_attributes_count is the number of attributes that were discarded. Attributes
+  // can be discarded because their keys are too long or because there are too many
+  // attributes. If this value is 0, then no attributes were dropped.
+  uint32 dropped_attributes_count = 5;
+
+  // This can be original pprof or jfr encoded profile.
+  bytes original_payload = 7;
+
+  // this is temporary until we pick one of these formats
+  // fields from one of these will be embedded in Profile message:
+  oneof alternative_profile {
+    opentelemetry.proto.profiles.v1.alternatives.pprof.Profile pprof = 8;
+    opentelemetry.proto.profiles.v1.alternatives.normalized.Profile normalized = 9;
```

Comment:
I'm waiting to see some benchmark, but I do think the `normalized` is the way to go.

With a lot of samples (we can expect ~10k per profiles) it avoid duplications. It will most likely have a smaller memory footprint and faster marshalling/unmarshalling.

It's also closer to pprof which is quite popular for profiling data.

Reply:
@cyriltovena Explain the math behind that 10k, please?

Reply:
@christos68k Yeah, pprof was included as a reference so that we'd be able to easily compare it to other ones. Long term plan is to keep just one of these 3 formats, and based on the discussions so far it seems like it will be based on the `normalized` version.